routes/user.playlist.js

```diff
@@ -252,139 +255,210 @@ router.route('/:playlist_name/export3')
+                color: '000000'
+            };
+            var margin = stanzasPerSlide === 1 ? 175 : 50;
+            for (var x = 0; x < songs.length; x++) {
+                for (var j = 0; j < songs[x].lyric[i].length; j++) {
+                    //print the lyric line for each translation
+                    pObj = slide.addText(songs[x].lyric[i][j], {
+                        x: 'c', //x position
+                        y: margin, //y position
+                        cx: '100%', //width
+                        cy: 40,
+                        font_size: 40,
+                        align: 'center',
+                        color: {
                             type: 'solid',
-                            color: '000000'
-                        };
-                        var margin = 50;
-                        for (var x = 0; x < songObj.songs.length; x++) {
-                            for (var j = 0; j < songObj.songs[x].lyric[i].length; j++) {
-                                //print the lyric line for each translation
-                                pObj = slide.addText(songObj.songs[x].lyric[i][j], {
-                                    x: 'c', //x position
-                                    y: margin, //y position
-                                    cx: '100%', //width
-                                    cy: 40,
-                                    font_size: 40,
-                                    align: 'center',
-                                    color: {
-                                        type: 'solid',
-                                        color: 'ffffff'
-                                    }
-                                });
-                                margin += 50;
-                            }
-                            margin += 150
+                            color: 'ffffff'
                         }
-
-                    }
-                })
-                callback();
+                    });
+                    margin += 50;
+                }
+                margin += 150
             }
-
-
-            function finalize() {
-                var out = fs.createWriteStream(filename + '.pptx');
-                out.on('error', function(err) {
-                    console.log(err);
-                });
-                res.writeHead(200, {
-                    "Content-Type": "application/vnd.openxmlformats-officedocument.presentationml.presentation",
-                    'Content-disposition': 'attachment; filename=' + filename + '.pptx'
-                });
-                pptx.generate(res);
+        }
+        var out = fs.createWriteStream(filename + '.pptx');
+        pptx.generate(out, {
+            'finalize': function(written) {
+                console.log('Finish to create a PowerPoint file.\nTotal bytes created: ' + written + '\n');
+                files.push(filename)
+                if (index === arr.length - 1) {
+                    setTimeout(() => {
```

Comment:
How do you know the timeout will be enough time? Probably need to wire up the onend event on the zip stream instead...there should be a way to make this work without using timeouts.